src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2085,16 +2085,19 @@ class RideConstructionWindow final : public Window
             }
 
             int32_t slope, bank;
+            bool buildDirectionIsDiagonal;
```

Comment:
For extra defensiveness, let’s initialise this to `false`.

Reply:
What would that defend against? It is always written to before it is read.

Reply:
It is now, but if the if-else conditions change, it won’t be. And it would most likely go unnoticed.